### PR TITLE
added `database` and `user` to the current and slow queries API return values

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Add attributes `database` and `user` when tracking current and slow AQL queries.

`database` contains the name of the database the query is/was running in, `user` contains the name of the user that started the query. These attributes will be returned in addition when calling the APIs for current and slow query inspection:

* GET `/_api/query/current` and `require("arangodb/aql/queries").current()`
* GET `/_api/query/slow` and `require("arangodb/aql/queries").slow()`

The "slow query" log message has also been augmented to contain the database name and the user name.

The `user` attribute is now also displayed in the web interface in the "Running queries" and "Slow queries" views.

- [ ] :hankey: Bugfix 
- [x] :pizza: New feature 
- [ ] :hammer: Refactoring 
- [x] :book: CHANGELOG entry made
- [x] :muscle: The behavior in this PR was *manually tested*
- [x] :computer: The behavior change can be verified via automatic tests

#### Backports:

- [ ] No backports required
- [x] Backports required for: 3.6

### Testing & Verification

- [x] This PR adds tests that were used to verify all changes:
  - [x] Added new **integration tests** (i.e. in shell_server / shell_client)

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11606/

### CODE REVIEW

**Comment by jsteemann** [comment]
Tests blue

File: arangod/Aql/QueryContext.cpp [hunk 1]
```diff
@@ -57,6 +58,9 @@ QueryContext::QueryContext(TRI_vocbase_t& vocbase)
   if (!AqlFeature::lease()) {
     THROW_ARANGO_EXCEPTION(TRI_ERROR_SHUTTING_DOWN);
   }
+
+  // store name of user that started the query
+  _user = ExecContext::current().user();
```
Comment (graetzer) [inline arangod/Aql/QueryContext.cpp:63]: I think this should go into the actual query

File: arangod/Aql/QueryContext.h [hunk 2]
```diff
@@ -82,6 +82,9 @@ class QueryContext {
 
   /// @brief return the names of collections used in the query
   std::vector<std::string> collectionNames() const;
+
+  /// @brief return the user that started the query
+  std::string const& user() const { return _user; }
   
   /// warnings access is thread safe
   QueryWarnings& warnings() { return _warnings; }
```
Comment (graetzer) [inline arangod/Aql/QueryContext.h:90]: make it perhaps virtual and return an empty string by default

**Review by graetzer — APPROVED** [review]

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11613/

**Review by graetzer — APPROVED** [review]

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11616/

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/11622/

**Comment by jsteemann** [comment]
Tests blue

File: arangod/Aql/Query.cpp [hunk 3]
```diff
@@ -149,6 +149,11 @@ Query::Query(std::shared_ptr<transaction::Context> const& ctx,
 
   _resourceMonitor.setMemoryLimit(_queryOptions.memoryLimit);
   _warnings.updateOptions(_queryOptions);
+
+  // store name of user that started the query
+  if (!ServerState::instance()->isDBServer()) {
+    _user = ExecContext::current().user();
```
Comment (graetzer) [inline arangod/Aql/Query.cpp:155]: User is empty anyway no need to check

**Review by graetzer — APPROVED** [review]

File: arangod/Aql/Query.cpp (hunk 3 above)
Comment (jsteemann) [reply arangod/Aql/Query.cpp:155]: Why? If this is running on coordinator or single server, we _should_ have a user set in most case.

Comment (jsteemann) [reply arangod/Aql/Query.cpp:155]: or are you suggesting we should assign to the variable unconditionally?

File: arangod/Aql/Query.cpp [hunk 4]
```diff
@@ -149,6 +149,11 @@ Query::Query(std::shared_ptr<transaction::Context> const& ctx,
...
+  }
```
Comment (jsteemann) [inline arangod/Aql/Query.cpp:156]: Is this your suggestion?
```suggestion
  // store name of user that started the query
  _user = ExecContext::current().user();
```

**Comment by graetzer** [comment]
Yes

**Comment by KVS85** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/11738/

**Comment by KVS85** [comment]
Tests blue.

**Review by KVS85 — APPROVED** [review]

**Pull request merged**